cumulusci/tasks/robotframework/lint.py

```diff
@@ -0,0 +1,207 @@
+
+        + example.robot
+        W: 2, 0: No suite documentation (RequireSuiteDocumentation)
+        E: 30, 0: No testcase documentation (RequireTestDocumentation)
+
+    To see a list of all configured options, set the 'list' option to True:
+
+        cci task run robot_list -o list True
+
+    """
+
+    task_options = {
+        "configure": {
+            "description": "List of rule configuration values, in the form of rule:args.",
+            "default": None,
+        },
+        "ignore": {
+            "description": "List of rules to ignore. Use 'all' to ignore all rules",
+            "default": None,
+        },
+        "error": {
+            "description": "List of rules to treat as errors. Use 'all' to affect all rules.",
+            "default": None,
+        },
+        "warning": {
+            "description": "List of rules to treat as warnings. Use 'all' to affect all rules.",
+            "default": None,
+        },
+        "list": {
+            "description": "If option is True, print a list of known rules instead of processing files.",
+            "default": None,
+        },
+        "path": {
+            "description": "The path to one or more files or folders. "
+            "If the path includes wildcard characters, they will be expanded. "
+            "If not provided, the default will be to process all files "
+            "under robot/<project name>",
+            "required": False,
+        },
+    }
+
+    def _validate_options(self):
+        super(RobotLint, self)._validate_options()
+        self.options["list"] = process_bool_arg(self.options.get("list", False))
+
+        self.options["path"] = process_list_arg(self.options.get("path", None))
+        self.options["ignore"] = process_list_arg(self.options.get("ignore", []))
+        self.options["warning"] = process_list_arg(self.options.get("warning", []))
+        self.options["error"] = process_list_arg(self.options.get("error", []))
+        self.options["configure"] = process_list_arg(self.options.get("configure", []))
+
+        if self.options["path"] is None:
+            self.options["path"] = [
+                "robot/{}".format(self.project_config.project["name"])
+            ]
+
+    def _run_task(self):
+        linter = CustomRfLint(task=self)
+        result = 0
+
+        if self.options["list"]:
+            linter.run(["--list"])
+
+        else:
+            files = self._get_files()
+            args = self._get_args()
+
+            # the result is a count of the number of errors,
+            # though I don't think the caller cares.
+            result = linter.run(args + sorted(files))
+
+        # result is the number of errors detected
+        if result > 0:
+            message = (
+                "1 error was detected"
+                if result == 1
+                else "{} errors were detected".format(result)
+            )
+            raise CumulusCIFailure(message)
+
+    def _get_files(self):
```

Comment:
Another approach might be:

```python
from pathlib import Path
def _get_files(self):
    cwd = Path('.')
    robot_files = set(cwd.glob('**/*.robot'))
    resource_files = set(cwd.glob('**/*.resource'))
    return {str(file) for file in robot_files.union(resource_files) if file.is_file()}
```

Reply:
Yeah, I thought about using pathlib, but it's only supported on 3.x. Don't we still require support for 2.7?

Reply:
Correct.

Reply:
I guess I thought we had `pathlib2` in the requirements for some reason. 🤷‍♂